File: webui/src/test/scala/com/taskadapter/webui/config/TaskFieldsMappingFragmentTest.scala
package com.taskadapter.webui.config

import com.taskadapter.connector.definition.FieldMapping
import com.taskadapter.model.{CreatedOn, UpdatedOn}
import com.taskadapter.webui.Page
import com.taskadapter.webui.uiapi.ConfigsTempFolder
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

import java.util

@RunWith(classOf[JUnitRunner])
class TaskFieldsMappingFragmentTest extends FunSpec with Matchers with ConfigsTempFolder {
  it("empty default value for non-string-based field is converted to empty string") {
    val f = new TaskFieldsMappingFragment(Page.MESSAGES,
      util.Arrays.asList(CreatedOn), Page.MESSAGES, "JIRA",
      util.Arrays.asList(CreatedOn, UpdatedOn), Page.MESSAGES, "Redmine",
      Seq(FieldMapping(CreatedOn, CreatedOn, true, null))
    )

    f.getElements.size shouldBe 1
    f.getElements.head.defaultValue shouldBe ""
  }
}

File: webui/src/test/java/com/taskadapter/webui/config/TaskFieldsMappingFragmentTest.java
package com.taskadapter.webui.config;

import com.taskadapter.common.ui.FieldMapping;
import com.taskadapter.model.AllFields;
import com.taskadapter.webui.Page;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class TaskFieldsMappingFragmentTest {
    @Test
    public void emptyDefaultValueForNonStringBasedFieldIsConvertedToEmptyString() {
        var f = new TaskFieldsMappingFragment(Page.MESSAGES,
                Arrays.asList(AllFields.createdOn()), Page.MESSAGES, "JIRA",
                Arrays.asList(AllFields.createdOn(), AllFields.updatedOn()), Page.MESSAGES, "Redmine",
                List.of(new FieldMapping(AllFields.createdOn(), AllFields.createdOn(), true, null))
        );

        assertThat(f.getElements()).hasSize(1);
        assertThat(f.getElements().get(0).getDefaultValue()).isEqualTo("");
    }
}
